divideconquer/BinarySearch.java

```diff
@@ -0,0 +1,68 @@
+import java.io.*;
+import java.util.*;
+
+public class BinarySearch {
+
+    static int binarySearch(int[] a, int x, int left, int right) {
+        int mid = (right-left)/2;
+        if(left<right)
+        {
+        if(a[mid] ==x)
+        return mid;
+        else if(a[mid]>x)
+        return binarySearch(a, x, left, mid);
+        else if(a[mid]<x)
+        return binarySearch(a, x, mid+1, right);
+        else
+        return -1;}
+        else if(left ==right)
+        {if(a[mid] ==x)
+        return mid;}
+	return -1;
+    }
+
+    public static void main(String[] args) {
+        FastScanner scanner = new FastScanner(System.in);
+        int n = scanner.nextInt();
+        int[] a = new int[n];
+        for (int i = 0; i < n; i++) {
```

Comment:
add white space between declaration/assignment and logic
i.e.
`int[] a = new int[n];`
` `
`for (int i = 0; i < n; i++) {`

Reply:
Ok I am on it